In [1]:
import pandas as pd 
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
data=pd.read_csv(r"C:\Users\DELL\Downloads\Advertising.csv")
data

,Unnamed: 0,TV,Radio,Newspaper,Sales
0,1,230.1,37.8,69.2,22.1
1,2,44.5,39.3,45.1,10.4
2,3,17.2,45.9,69.3,9.3
3,4,151.5,41.3,58.5,18.5
4,5,180.8,10.8,58.4,12.9
...,...,...,...,...,...
195,196,38.2,3.7,13.8,7.6
196,197,94.2,4.9,8.1,9.7
197,198,177.0,9.3,6.4,12.8
198,199,283.6,42.0,66.2,25.5


In [2]:
#step1: independent and dependent variable creation
X=data[['TV','Radio','Newspaper']] #contains all inputs
y=data.Sales  #target

In [3]:
from sklearn.model_selection import train_test_split
X_train,X_test,y_train,y_test=train_test_split(X,y,test_size=0.2,random_state=4)

In [4]:
print(len(X_test))
print(len(X_train))

40
160


In [7]:
#step3: creating a model
from sklearn.linear_model import LinearRegression
from sklearn.metrics import r2_score,mean_squared_error,mean_absolute_error,root_mean_squared_error
LR=LinearRegression() #object creation
LR.fit(X_train,y_train) #training of linear regression
y_predict=LR.predict(X_test)
print("R2: ",r2_score(y_test,y_predict))

R2:  0.9085774752313166


In [8]:
print("MSE: ",mean_squared_error(y_test,y_predict))
print("MAE: ",mean_absolute_error(y_test,y_predict))
print("RMSE: ",root_mean_squared_error(y_test,y_predict))

MSE:  2.217151072541898
MAE:  1.2114892606469108
RMSE:  1.4890100982001089


Remove 'Newspaper' column :

In [9]:
X=data[['TV','Radio']] 
y=data.Sales

In [10]:
from sklearn.model_selection import train_test_split
X_train,X_test,y_train,y_test=train_test_split(X,y,test_size=0.2,random_state=4)

In [11]:
from sklearn.linear_model import LinearRegression
from sklearn.metrics import r2_score,mean_squared_error,mean_absolute_error,root_mean_squared_error
LR=LinearRegression() #object creation
LR.fit(X_train,y_train) #training of linear regression
y_predict=LR.predict(X_test)
print("R2: ",r2_score(y_test,y_predict))

R2:  0.9086623887492478


In [12]:
print("MSE: ",mean_squared_error(y_test,y_predict))
print("MAE: ",mean_absolute_error(y_test,y_predict))
print("RMSE: ",root_mean_squared_error(y_test,y_predict))

MSE:  2.215091775909797
MAE:  1.210808620492933
RMSE:  1.4883184390142443


From the simple linear regression:
TV  0.46
Radio  0.37
Newspaper  0.117

From the multiple linear regression:
TV+Radio  0.908  ->we are using only 66% data
TV+Radio+Newspaper  0.908  ->we are using more than 66% data
->we are getting same accuracy in both ways, we use TV+Radio.
->computational complexity in ML is used for this decision.